### PR TITLE
First step towards a ring of polynomials: left multiplication by a scalar

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: quantities/tuples.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// This struct has a |Type| member which is a tuple obtained by applying
+// |Transform| to each element type in |Tuple| (which must be a tuple or an array
+// or a pair).
+template<template<typename> class Transform,
+         typename Tuple,
+         typename = std::make_integer_sequence<int, std::tuple_size_v<Tuple>>>
+struct ApplyGenerator;
+
+// This struct has a |Type| member which is a tuple made of successive
+// derivatives of |Value| with respect to |Argument| from 0 to |n|, exclusively.
```
Comment (eggrobin) [inline quantities/tuples.hpp]: [...] which is an n-element tuple of successive derivatives of |Value| with respect to |Argument|; the first element is |Value|.

File: numerics/polynomial_body.hpp [hunk 2]
```diff
@@ -14,6 +15,31 @@ namespace internal_polynomial {
...
+  Multiply(Scalar const& left, Tuple const& right);
+};
+
+template<typename Scalar, typename Tuple, int... indices>
+constexpr auto
+TupleArithmetic<Scalar, Tuple, std::integer_sequence<int, indices...>>::
+    Multiply(Scalar const& left, Tuple const& right) ->
+    typename Apply<ScalarLeftMultiplier, Tuple> {
+  return {left * std::get<indices>(right)...};
+}
```
Comment (eggrobin) [inline numerics/polynomial_body.hpp]: Would `std::apply` work?

Comment (eggrobin) [reply numerics/polynomial_body.hpp]: Nevermind, that's completely unrelated.

File: quantities/tuples.hpp (hunk 1 above)
Comment (pleroy) [reply quantities/tuples.hpp]: Done.

File: numerics/polynomial_body.hpp (hunk 2 above)
Comment (pleroy) [reply numerics/polynomial_body.hpp]: No.

**Pull request merged**